/*
   Copyright 2022-2024 mkckr0 <https://github.com/mkckr0>

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

	   http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
*/

syntax = "proto3";

package io.github.mkckr0.audio_share_app.pb;

message AudioFormat
{
   enum Encoding {
      ENCODING_INVALID = 0;
      ENCODING_PCM_F32 = 1;   // PCM encoding are all little endian
      ENCODING_PCM_U8 = 2;
      ENCODING_PCM_U16 = 3;
      ENCODING_PCM_U24 = 4;
      ENCODING_PCM_U32 = 5;
   }

	Encoding encoding = 1;
	int32 channels = 2;
	int32 sample_rate = 3;
}
